ci/test_supported_plugins.rb

```diff
@@ -0,0 +1,253 @@
+# encoding: utf-8
+
+# Test the 
+# - build (rspec), 
+# - packaging (gem build)
+# - and deploy (bin/logstash-plugin install)
+# of a plugins inside the current Logstash, using its JRuby
+# Usage example:
+# bin/ruby ci/test_supported_plugins.rb -p logstash-integration-jdbc
+# bin/ruby ci/test_supported_plugins.rb -t tier1 -k input,codec,integration
+
+require "open3"
+require "set"
+require 'optparse'
+
+ENV['LOGSTASH_PATH'] = Dir.pwd
+ENV['LOGSTASH_SOURCE'] = '1'
+
+logstash_plugin_cli = ENV['LOGSTASH_PATH'] + "/bin/logstash-plugin"
+
+# it has to be out of logstash local close else plugins' Gradle script
+# would interfere with Logstash's one
+base_folder = "/tmp"
+plugins_folder = File.join(base_folder, "plugin_clones")
+unless File.directory?(plugins_folder)
+  Dir.mkdir(plugins_folder)
+end
+
+class Plugin
+  attr_reader :plugins_folder, :plugin_name, :plugin_base_folder
+
+  def initialize(plugins_folder, plugin_name)
+    @plugin_name = plugin_name
+    @plugins_folder = plugins_folder
+    @plugin_base_folder = "#{plugins_folder}/#{plugin_name}"
+  end
+
+  def git_clone
+    if File.directory?(plugin_name)
+      puts "#{plugin_name} already cloned locally"
+      return
```

Comment:
should we `git pull` in this case? do we need to worry about "dirty" repositories with untracked or modified files?

Reply:
Yes, great idea thanks. Implemented with commit e23677bbbaf2baa2c7e6372f3985bdaa44c38854